### ISSUE TITLE
Multivariate normal log-likelihood?

### ISSUE THREAD

**Issue by nguyenhoaiThanhbk2811** [aloctavodia/BAP#19]
Hi you,
Now, I am using Multivariate normal log-likelihood to find the posterior distribution but I don't know how to use Multivariate normal log-likelihood in pymc3.

I have an equation of input and output:

 mu = a * q0 ** 2 + b * q0 * q1 + c * q1 ** 2 - d * q0 * q2 + e * q0 - f * q1 * q2 + g * q2 ** 2 - h * q1 - i * q2 + j

where:
a = -1.2422069376608758
b = 0.010979502286122458
c = 0.005065108695132867
d = 0.0003998832861218647
e = 0.16015398278991752
f = 0.00013362145005939259
g = 2.0112165363058473e-06
h = 0.0020280971401678734
i = 4.0276945348026916e-05
j = 0.01657469140501759

with 
q0= [0.05675, 0.05934, 0.05633, 0.0557 , 0.05702, 0.06401, 0.06322, 0.06571, 0.06099, 0.05832, 0.06196, 0.06463, 0.05507, 0.06351, 0.06287, 0.06122, 0.05407, 0.05985, 0.05774,0.06015]

q1 = [0.9486, 0.9095, 0.9856, 0.9318, 1.0477, 1.0489,1.0663, 0.9184, 0.9646, 1.0345, 1.0168, 1.0565, 0.9727, 0.9907, 0.9277, 0.9548, 1.0933, 1.0751,1.0026, 1.0231]

q2= [51.813, 54.279, 52.659, 51.197 , 46.629, 49.791, 48.581, 54.799, 46.413, 47.078, 52.367, 48.204, 50.389, 45.402, 47.893, 50.796 , 49.332, 53.323, 53.713, 45.757]
mu = [0.0204232,0.0205054,0.0204971,0.0204463,0.0206686,0.0206678,0.0206883,0.0204627,0.020426,0.0206532,0.0206322,0.020677,0.0204431,0.0205319,0.0204508,0.0204115,0.020721,0.0206988,0.0206179,0.0206418]
from the each variable value above I estimate the surrogate model look like mu equation. and I also have prior distribution of (q0,q1,q2) is normal distribution.
q0 Normal (0.05,0.06)
q1 Normal (1.0,0.1)
q2 Normal (50,5)

How to fin the posterior distribution p(mu| (q0,q1,q2)).

I hope you to help me.

Thanks you.

**Comment by aloctavodia** [comment]
Hi @nguyenhoaiThanhbk2811

As with your previous questions you are already proving the values of the parameters so it is not clear what are you trying to infer here. Please explain what is your data and what are the parameters you are trying to find and if possible try to write at least part of the model you want to build and indicate which part you do not know how to write.

**Comment by nguyenhoaiThanhbk2811** [comment]
`import pymc3 as pm
import numpy as np
import chaospy as cp
import pandas as pd
import scipy.stats as stats
from sklearn.preprocessing import StandardScaler
import matplotlib.pyplot as plt
import seaborn as sns

nodes1 = np.array([0.05675, 0.05934, 0.05633, 0.0557 , 0.05702, 0.06401, 0.06322, 0.06571, 0.06099, 0.05832, 0.06196, 0.06463, 0.05507, 0.06351, 0.06287, 0.06122, 0.05407, 0.05985, 0.05774,0.06015])
nodes2 = np.array([0.9486, 0.9095, 0.9856, 0.9318, 1.0477, 1.0489,1.0663, 0.9184, 0.9646, 1.0345, 1.0168, 1.0565, 0.9727, 0.9907, 0.9277, 0.9548, 1.0933, 1.0751,1.0026, 1.0231])
nodes3 = np.array([51.813, 54.279, 52.659, 51.197 , 46.629, 49.791, 48.581, 54.799, 46.413, 47.078, 52.367, 48.204, 50.389, 45.402, 47.893, 50.796 , 49.332, 53.323, 53.713, 45.757])


#print (nodes1)
distribution = cp.J(cp.Uniform(0.054, 0.066), cp.Uniform(0.9, 1.1), cp.Uniform( 45, 55))
#cp.seed (20)
#samples = distribution.sample(20,"L")



#print (samples)


nodes = np.vstack([nodes1, nodes2, nodes3])
evals = np.array([0.0204232,0.0205054,0.0204971,0.0204463,0.0206686,0.0206678,0.0206883,0.0204627,0.020426,0.0206532,0.0206322,0.020677,0.0204431,0.0205319,0.0204508,0.0204115,0.020721,0.0206988,0.0206179,0.0206418])


#distribution2=cp.Uniform(0.023991,0.0244017)
polynomial_expansion = cp.orth_ttr(2, distribution)
approx = cp.fit_regression(polynomial_expansion, nodes, evals)  # surrogate model




print (approx)
# bayesian


a = -1.2422069376608758
b = 0.010979502286122458
c = 0.005065108695132867
d = 0.0003998832861218647
e = 0.16015398278991752
f = 0.00013362145005939259
g = 2.0112165363058473e-06
h = 0.0020280971401678734
i = 4.0276945348026916e-05
j = 0.01657469140501759


basic_model = pm.Model()

with basic_model:
    q0 = pm.Normal('q0', 0.05, 0.06)
    q1 = pm.Normal('q1', 1.0, 0.1)
    q2 = pm.Normal('q2', 50, 5)
    mu = a * q0 ** 2 + b * q0 * q1 + c * q1 ** 2 - d * q0 * q2 + e * q0 - f * q1 * q2 + g * q2 ** 2 - h * q1 - i * q2 + j
    pm.Normal('observed', mu, 0.05, observed=evals)
    trace = pm.sample(10000, cores =1)
    pm.traceplot(trace)
    k = pm.summary(trace).round(2)
    tracedf1 = pm.trace_to_dataframe (trace, varnames = ['q0','q1','q2'])
    sns.pairplot(tracedf1 )
print (k)
plt.show()`
this is my code.
But in my code just used the pm.Normal ('observed',mu,0.05,observed=evals). I want to Use the multivariate Normal. But I don't Know what thing I need to use multivariate Normal.
